File: OldBodyYoungSoul/samples/object_extansion/simple_extend.js
(function(){
  var objectToExtend = {};
  var notExtended_object = {};
  
  objectToExtend.say_something = function ()
  {
    console.log("something from objectToExtend");
  }

  if(typeof(objectToExtend.say_something) === 'function') objectToExtend.say_something();
  if(typeof(notExtended_object.say_something) === 'function') notExtended_object.say_something();

})();

File: OldBodyYoungSoul/en/samples/object_extansion/simple_extend.rb
object_to_extend = Object.new
not_extented_object = Object.new

object_to_extend.instance_eval do 
  def say_something
    puts 'somthing on object to extend'
  end
end

object_to_extend.say_something if object_to_extend.respond_to? 'say_something'
not_extented_object.say_something if not_extented_object.respond_to? 'say_something'
